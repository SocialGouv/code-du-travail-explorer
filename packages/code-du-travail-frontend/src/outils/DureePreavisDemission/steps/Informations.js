import React from "react";
import data from "@cdt/data...preavis-demission/data.json";

import { SelectQuestion } from "../../common/SelectQuestion";
import { SectionTitle } from "../../common/stepStyles";
import {
  getOptions,
  getPastQuestions,
  getNextQuestionKey,
  filterSituations,
  getSituationsFor,
  getFormProps
} from "../../common/situations.utils";

const { questions, situations: allSituations } = data;
const questionsMap = questions.reduce(
  (state, { name, question }) => ({ ...state, [name]: question }),
  {}
);

function StepInformations({ form }) {
  const { values } = form.getState();
  const { ccn, criteria = {} } = values;
  const idcc = ccn ? ccn.num : "0000";

  const initialSituations = getSituationsFor(allSituations, { idcc });
  const possibleSituations = filterSituations(initialSituations, criteria);
  const nextQuestionKey = getNextQuestionKey(possibleSituations, criteria);
  const nextQuestionOptions = getOptions(possibleSituations, nextQuestionKey);
  const pastQuestions = getPastQuestions(initialSituations, criteria);

  return (
    <>
      <SectionTitle>Informations sur le statut du salari√©</SectionTitle>
      {pastQuestions.map(([key, answers]) => (
        <SelectQuestion
          key={key}
          name={`criteria.${key}`}
          options={answers}
          label={questionsMap[key]}
          onChange={() =>
            form.batch(() => {
              getFormProps({
                key,
                criteria,
                pastQuestions
              }).forEach(key => form.change(`criteria.${key}`, undefined));
            })
          }
        />
      ))}
      {nextQuestionKey && nextQuestionOptions && (
        <>
          <SelectQuestion
            name={`criteria.${nextQuestionKey}`}
            label={questionsMap[nextQuestionKey]}
            options={nextQuestionOptions}
          />
        </>
      )}
    </>
  );
}

export { StepInformations };
