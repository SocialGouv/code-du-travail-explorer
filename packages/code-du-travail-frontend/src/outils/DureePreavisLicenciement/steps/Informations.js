import React from "react";
import { YesNoQuestion } from "../../common/YesNoQuestion";

function validate() {
  const errors = {};
  return errors;
}

function StepInformations({ form }) {
  const data = form.getState().values;
  return (
    <>
      <YesNoQuestion
        name="fauteGrave"
        label="S'agit-il d'un licenciement pour faute grave ou lourde ?"
      />
      {data.fauteGrave === false && (
        <>
          <YesNoQuestion
            name="travailleurHandicape"
            label="Le salarié concerné est-il reconnu en tant que travailleur handicapé ?"
          />
        </>
      )}
    </>
  );
}
StepInformations.validate = validate;
export { StepInformations };
