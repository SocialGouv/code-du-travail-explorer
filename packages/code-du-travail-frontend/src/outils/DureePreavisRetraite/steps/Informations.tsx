import React, { useEffect } from "react";

import { TextQuestion } from "../../common/TextQuestion";
import { WizardStepProps } from "../../common/type/WizardType";
import { usePublicodes } from "../../publicodes";
import { transformInfoCcn } from "../../publicodes/TransformInfoCcn";

function Informations({ form }: WizardStepProps): JSX.Element {
  const publicodesContext = usePublicodes();

  const excludedRules = [
    "contrat salarié - ancienneté",
    "contrat salarié - convention collective",
    "contrat salarié - mise à la retraite",
  ];
  useEffect(() => {
    publicodesContext.setSituation(transformInfoCcn(form.getState().values));
  }, [form]);

  return (
    <>
      <>
        {publicodesContext.situation
          .filter((item) => !excludedRules.includes(item.name))
          .map((item) => {
            return (
              <TextQuestion
                key={item.name}
                name={item.name}
                label={item.rawNode.question}
                validate={undefined}
                placeholder="0"
              />
            );
          })}
      </>
      <>
        {publicodesContext.missingArgs
          .filter((item) => !excludedRules.includes(item.name))
          .map((item) => {
            return (
              <TextQuestion
                key={item.name}
                name={item.name}
                label={item.rawNode.question}
                validate={() => true}
                placeholder="0"
              />
            );
          })}
      </>
    </>
  );
}

export { Informations };
