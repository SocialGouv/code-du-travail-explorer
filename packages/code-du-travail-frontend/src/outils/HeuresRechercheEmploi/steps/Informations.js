import React from "react";
import data from "@cdt/data...simulateurs/heure-recherche-emploi.data.json";

import { SelectQuestion } from "../../common/SelectQuestion";
import { SectionTitle } from "../../common/stepStyles";
import {
  getOptions,
  getPastQuestions,
  getNextQuestionKey,
  filterSituations,
  getSituationsFor,
  getFormProps
} from "../../common/situations.utils";

const { questions, situations: allSituations } = data;
const questionsMap = questions.reduce(
  (state, { name, question }) => ({ ...state, [name]: question }),
  {}
);

const criteriaOrder = questions.map(({ name }) => name);

function StepInformations({ form }) {
  const { values } = form.getState();
  const { ccn, criteria = {} } = values;
  const idcc = ccn ? ccn.convention.num : 0;

  const initialSituations = getSituationsFor(allSituations, { idcc });
  const possibleSituations = filterSituations(initialSituations, criteria);
  const nextQuestionKey = getNextQuestionKey(
    possibleSituations,
    criteriaOrder,
    criteria
  );
  const nextQuestionOptions = getOptions(possibleSituations, nextQuestionKey);
  const pastQuestions = getPastQuestions(
    initialSituations,
    criteriaOrder,
    criteria
  );

  return (
    <>
      <SectionTitle>Statut du salari√©</SectionTitle>
      {pastQuestions.map(([key, answers]) => (
        <SelectQuestion
          key={key}
          name={`criteria.${key}`}
          options={answers}
          label={questionsMap[key]}
          onChange={() =>
            form.batch(() => {
              getFormProps({
                key,
                criteria,
                pastQuestions
              }).forEach(key => form.change(`criteria.${key}`, undefined));
            })
          }
        />
      ))}
      {nextQuestionKey && nextQuestionOptions && (
        <>
          <SelectQuestion
            name={`criteria.${nextQuestionKey}`}
            label={questionsMap[nextQuestionKey]}
            options={nextQuestionOptions}
          />
        </>
      )}
    </>
  );
}

export { StepInformations };
