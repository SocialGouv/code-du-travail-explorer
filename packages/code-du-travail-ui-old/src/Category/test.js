import React from "react";
import { render } from "@testing-library/react";
import { Category } from ".";

describe("<Category />", () => {
  test("should render with default props", () => {
    const { container } = render(
      <Category
        title="title"
        text="texte"
        icon="/public/icons/remuneration.svg"
      />
    );
    expect(container).toMatchSnapshot();
  });
  test("should render small", () => {
    const { container } = render(
      <Category
        small
        title="title"
        text="texte"
        icon="/public/icons/remuneration.svg"
      />
    );
    expect(container).toMatchSnapshot();
  });
});
