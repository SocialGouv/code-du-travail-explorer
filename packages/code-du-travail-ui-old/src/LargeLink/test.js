import React from "react";
import { render } from "@testing-library/react";
import LargeLink from ".";

describe("<LargeLink />", () => {
  test("should render", () => {
    const { container } = render(<LargeLink>Ok</LargeLink>);
    expect(container).toMatchSnapshot();
  });
  test("should render light", () => {
    const { container } = render(<LargeLink variant="light">Ok</LargeLink>);
    expect(container).toMatchSnapshot();
  });
  test("should render highlighted", () => {
    const { container } = render(<LargeLink variant="highlight">Ok</LargeLink>);
    expect(container).toMatchSnapshot();
  });
});
