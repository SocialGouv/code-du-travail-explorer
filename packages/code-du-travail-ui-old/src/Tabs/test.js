import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Tabs from ".";

const items = [
  {
    tab: "Tab 1",
    panel: "This panel can contain nodes"
  },
  {
    tab: "Tab 2",
    panel: "These tabs are not 'controlled'"
  }
];

describe("<Tabs />", () => {
  it("renders", () => {
    const { container } = render(<Tabs data={items} />);
    expect(container).toMatchSnapshot();
  });
  it("calls the right onSelect callback", () => {
    const onSelect = jest.fn();
    const { container } = render(<Tabs onSelect={onSelect} data={items} />);
    const secondTab = container.querySelectorAll("li")[1];
    fireEvent.click(secondTab);
    expect(onSelect).toHaveBeenCalledTimes(1);
  });
});
