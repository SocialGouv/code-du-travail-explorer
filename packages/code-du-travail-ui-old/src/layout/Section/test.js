import React from "react";
import { render } from "@testing-library/react";
import Section from ".";

describe("<Section />", () => {
  test.each([["default"], ["white"], ["dark"], ["light"]])(
    "it renders a %s Section",
    variant => {
      const { container } = render(
        <Section variant={variant}>this is a Button {variant} </Section>
      );
      expect(container).toMatchSnapshot();
    }
  );
});
