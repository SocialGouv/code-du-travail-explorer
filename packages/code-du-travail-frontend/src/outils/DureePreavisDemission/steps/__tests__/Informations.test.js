import { StepInformations } from "../Informations";
import { renderForm } from "../../../../../test/renderForm";

describe("<StepInformations />", () => {
  it("should render", () => {
    const { container } = renderForm(StepInformations);
    expect(container).toMatchSnapshot();
  });
  it("should render catégorie", () => {
    const { container } = renderForm(StepInformations, {
      branche: "0044"
    });
    expect(container).toMatchSnapshot();
  });
  it("should render coefficient", () => {
    const { container } = renderForm(StepInformations, {
      branche: "0044",
      catégorie: "1 Employés"
    });
    expect(container).toMatchSnapshot();
  });
});
