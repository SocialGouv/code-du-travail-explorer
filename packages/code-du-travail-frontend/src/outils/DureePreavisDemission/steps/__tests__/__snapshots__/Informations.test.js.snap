// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepInformations /> should render 1`] = `
.c3 {
  position: absolute;
  background-color: #7598d6;
  border-radius: 0.6rem;
  top: 50%;
  left: 0;
  width: 0.4rem;
  height: 100%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c2 {
  position: relative;
  margin: 0;
  color: #2f3b6c;
  font-weight: normal;
  font-size: 2.8rem;
  font-family: "Merriweather",serif;
  line-height: 1.25;
  padding-left: 3.2rem;
  text-align: left;
}

.c0 {
  margin: 0 auto 1rem;
  text-align: left;
}

.c1 {
  margin-top: 3.2rem;
  margin-bottom: 2rem;
}

@media (max-width:600px) {
  .c2 {
    padding-left: 1.6rem;
  }
}

@media (max-width:600px) {
  .c2 {
    font-size: 2.2rem;
  }
}

<div>
  <header
    class="c0 c1"
  >
    <h2
      class="c2"
    >
      <div
        class="c3"
      />
      Informations sur le statut du salarié
    </h2>
  </header>
</div>
`;

exports[`<StepInformations /> should render catégorie 1`] = `
.c3 {
  position: absolute;
  background-color: #7598d6;
  border-radius: 0.6rem;
  top: 50%;
  left: 0;
  width: 0.4rem;
  height: 100%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c2 {
  position: relative;
  margin: 0;
  color: #2f3b6c;
  font-weight: normal;
  font-size: 2.8rem;
  font-family: "Merriweather",serif;
  line-height: 1.25;
  padding-left: 3.2rem;
  text-align: left;
}

.c0 {
  margin: 0 auto 1rem;
  text-align: left;
}

.c5 {
  display: block;
  margin-top: 2rem;
  margin-bottom: 1rem;
  font-size: 1.8rem;
  cursor: pointer;
}

.c6 {
  -webkit-flex: 1 1 70%;
  -ms-flex: 1 1 70%;
  flex: 1 1 70%;
  min-width: 400px;
  max-width: 100%;
  margin-top: 1rem;
  margin-right: 2rem;
}

.c4 {
  margin: 2rem 0;
}

.c1 {
  margin-top: 3.2rem;
  margin-bottom: 2rem;
}

@media (max-width:600px) {
  .c2 {
    padding-left: 1.6rem;
  }
}

@media (max-width:600px) {
  .c2 {
    font-size: 2.2rem;
  }
}

@media (max-width:600px) {
  .c5 {
    font-size: 1.6rem;
  }
}

<div>
  <header
    class="c0 c1"
  >
    <h2
      class="c2"
    >
      <div
        class="c3"
      />
      Informations sur le statut du salarié
    </h2>
  </header>
  <div
    class="c4"
  >
    <label
      class="c5"
      for="input-criteria.catégorie socio-professionnelle"
    >
      Quelle est la catégorie professionnelle du salarié ?
    </label>
    <select
      class="c6"
      id="input-criteria.catégorie socio-professionnelle"
      name="criteria.catégorie socio-professionnelle"
    >
      <option
        disabled=""
        value=""
      >
        ...
      </option>
      <option
        value="42| Collaborateurs"
      >
        Collaborateurs
      </option>
      <option
        value="48| Cadres"
      >
        Cadres
      </option>
    </select>
  </div>
</div>
`;

exports[`<StepInformations /> should render coefficient 1`] = `
.c3 {
  position: absolute;
  background-color: #7598d6;
  border-radius: 0.6rem;
  top: 50%;
  left: 0;
  width: 0.4rem;
  height: 100%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c2 {
  position: relative;
  margin: 0;
  color: #2f3b6c;
  font-weight: normal;
  font-size: 2.8rem;
  font-family: "Merriweather",serif;
  line-height: 1.25;
  padding-left: 3.2rem;
  text-align: left;
}

.c0 {
  margin: 0 auto 1rem;
  text-align: left;
}

.c5 {
  display: block;
  margin-top: 2rem;
  margin-bottom: 1rem;
  font-size: 1.8rem;
  cursor: pointer;
}

.c6 {
  -webkit-flex: 1 1 70%;
  -ms-flex: 1 1 70%;
  flex: 1 1 70%;
  min-width: 400px;
  max-width: 100%;
  margin-top: 1rem;
  margin-right: 2rem;
}

.c4 {
  margin: 2rem 0;
}

.c1 {
  margin-top: 3.2rem;
  margin-bottom: 2rem;
}

@media (max-width:600px) {
  .c2 {
    padding-left: 1.6rem;
  }
}

@media (max-width:600px) {
  .c2 {
    font-size: 2.2rem;
  }
}

@media (max-width:600px) {
  .c5 {
    font-size: 1.6rem;
  }
}

<div>
  <header
    class="c0 c1"
  >
    <h2
      class="c2"
    >
      <div
        class="c3"
      />
      Informations sur le statut du salarié
    </h2>
  </header>
  <div
    class="c4"
  >
    <label
      class="c5"
      for="input-criteria.catégorie socio-professionnelle"
    >
      Quelle est la catégorie professionnelle du salarié ?
    </label>
    <select
      class="c6"
      id="input-criteria.catégorie socio-professionnelle"
      name="criteria.catégorie socio-professionnelle"
    >
      <option
        disabled=""
        value=""
      >
        ...
      </option>
      <option
        value="42| Collaborateurs"
      >
        Collaborateurs
      </option>
      <option
        value="48| Cadres"
      >
        Cadres
      </option>
    </select>
  </div>
  <div
    class="c4"
  >
    <label
      class="c5"
      for="input-criteria.coefficient"
    >
      Quel est le coefficient hiérarchique du salarié ?
    </label>
    <select
      class="c6"
      id="input-criteria.coefficient"
      name="criteria.coefficient"
    >
      <option
        disabled=""
        value=""
      >
        ...
      </option>
      <option
        value="24| 700 à 750"
      >
        700 à 750
      </option>
      <option
        value="28| 800 à 830 inclus"
      >
        800 à 830 inclus
      </option>
    </select>
  </div>
</div>
`;
